pkg/utils/utils.go

```diff
@@ -84,3 +86,31 @@ func discardNil(ch chan error) chan error {
 	}()
 	return r
 }
+
+// IsWritable determines if a directory is writeable
+func IsWritable(path string) (bool, error) {
+	info, err := os.Stat(path)
+	if err != nil {
+		return false, err
+	}
+
+	if !info.IsDir() {
+		return false, errs.New("Path %s is not a directory", path)
+	}
+
+	// Check if the user bit is enabled in file permission
+	if info.Mode().Perm()&(1<<(uint(7))) == 0 {
+		return false, errs.New("Write permission bit is not set on this file for user")
+	}
+
+	var stat syscall.Stat_t
```

Comment:
All `syscall` things need to have appropriate build tags, because they are not cross-platform. Use os package instead (if possible), otherwise you need to write code for all other platforms.

Reply:
I made a function for windows but I wasn't sure how to do things like IFDEF in go

Reply:
You need to create separate files for each of the different platforms. You can see the full list in https://golang.org/doc/install/source#environment. For posix compatible code you probably can use an appropriate build-tag.

Reply:
Also before committing, it needs to be tested with Windows natively, Windows docker and Windows WSL.

Reply:
removed windows specific utils